pkg/cmd/server/start.go

```diff
@@ -794,33 +204,40 @@ func env(key string, defaultValue string) string {
 	return val
 }
 
-func expandImage(component, defaultValue string, latest bool) string {
-	template := defaultValue
-	if s, ok := imageComponentEnvExpander(component); ok {
-		template = s
-	}
-	value, err := variable.ExpandStrict(template, func(key string) (string, bool) {
-		switch key {
-		case "component":
-			return component, true
-		case "version":
-			if latest {
-				return "latest", true
-			}
-		}
-		return "", false
-	}, variable.Versions)
+func (cfg Config) RunEtcd() error {
+	etcdAddr, err := cfg.GetEtcdAddress()
 	if err != nil {
-		glog.Fatalf("Unable to find an image for %q due to an error processing the format: %v", err)
+		return err
+	}
+
+	etcdConfig := &etcd.Config{
+		BindAddr:     getHostOfHost(cfg.BindAddr.URL.Host),
```

Comment:
Actually, it looks like we're counting on etcd defaulting to binding on 4001 when we don't specify a port. Should probably have GetEtcdBindAddress() which does `net.JoinHostPort(cfg.BindAddr.Host, strconv.Itoa(cfg.EtcdAddr.DefaultPort))`